src/components/transaction/DataFormat.vue

```diff
@@ -21,46 +22,57 @@ export default defineComponent({
         },
     },
     setup(props) {
-        const { actionName, actionData, useColor } = toRefs(props);
-        const dataBox = ref(null);
+        const { actionName, actionData } = toRefs(props);
+        const dataBox = ref<null | HTMLElement>(null);
         const showOverflow = ref(false);
         const maxHeight = ref(57);
         const switchHeight = ref(20);
         const isOverflowing = ref(false);
         const transferData = computed(() => actionData.value as TransferData);
         const clientHeight = computed(() => dataBox.value?.clientHeight ?? 0);
+        let currentData = ref<string | unknown>(null);
+
+
+        function compareJsonObjects(obj1: unknown, obj2: unknown): boolean {
+            if (typeof obj1 !== 'object' || typeof obj2 !== 'object') {
+                return false;
+            }
+
+            const objectOne = obj1 as Record<string, unknown>;
+            const objectTwo = obj2 as Record<string, unknown>;
+
+            for (const key of Object.keys(obj1)) {
+                if (!objectOne[key]) {
+                    // this is an empty key
+                    // sometimes the answer includes them, sometimes it doesn't
+                    // so we won't compare them
+                    continue;
+                }
+
+                if (typeof objectOne[key] !== typeof objectTwo[key]) {
+                    return false;
+                }
 
-        function formatGeneralData(data: any): any[] {
-            var dict: any[] = [];
-            for (let key in data) {
-                if (data[key] instanceof Object) {
-                    if (Array.isArray(data[key])) {
-                        var keyValArray: any[] = [];
-                        for (let i = 0; i < data[key].length; i++) {
-                            if (data[key][i] instanceof Object) {
-                                keyValArray = keyValArray.concat(
-                                    formatGeneralData(data[key][i]),
-                                );
-                            }
-                        }
-                        if (keyValArray.length == 0) {
-                            dict.push({ key, value: JSON.stringify(data[key]) });
-                        } else {
-                            dict = dict.concat(keyValArray);
-                        }
-                    } else {
-                        var keyValArray: any[] = [];
-                        let formatData = formatGeneralData(data[key]);
-                        if (formatData.length > 0) {
-                            keyValArray = keyValArray.concat(formatGeneralData(data[key]));
-                            dict = dict.concat(keyValArray);
-                        }
+                if (objectOne[key] !== null && typeof objectOne[key] === 'object') {
```

Comment:
```suggestion
                if (typeof objectOne[key] === 'object') {
```
the first condition is redundant and covered by line 45 (never reaches this code if it _is_ `null`)

Reply:
I would rather we just remove the redundant condition. Including `!compareJsonObjects(objectOne[key], objectTwo[key])) || objectOne[key] !== objectTwo[key]` would make `compareJsonObjects` to be called every time, even when `objectOne[key]` is not an `object`.
We could change the order to avoid that, putting `objectOne[key] !== objectTwo[key]` comparison first, but I still feel that it just makes it harder to read. 

Would you be okay with this?

```
if (typeof objectOne[key] === 'object') {
    if (!compareJsonObjects(objectOne[key], objectTwo[key])) {
        return false;
    }
} else {
    if (objectOne[key] !== objectTwo[key]) {
        return false;
    }
}
```

Reply:
No, if that first condition itsn't met, it will short circuit and not evaluate the `compareObjects()` method